src/js/models/layer.js

```diff
@@ -310,7 +322,13 @@ define(function (require, exports, module) {
      * @param {object} layerEffect instance of a layer effect such as a Shadow
      */
     Layer.prototype.setLayerEffectByType = function (layerEffectType, layerEffectIndex, layerEffect) {
-        return this.setIn([layerEffectType + "s", layerEffectIndex], layerEffect);
+        if (!LayerEffect.TYPES.has(layerEffectType)) {
+            throw "Invalid layer effect type: " + layerEffectType;
```

Comment:
We should only ever throw error objects so that they can be handled (e.g., with `.message` and `.stack`) uniformly at a higher level. So, these should be changed to something like `throw new Error("Invalid layer...`